assets/large_language_models/rag/components/src/flow_creation.py

```diff
@@ -300,123 +330,178 @@ def main(args, ws, current_run, activity_logger: Logger):
+                json_stringify(top_prompts[1]), _CITATION_TEMPLATE, _USER_INPUT, is_chat
+            ),
+        )
         flow_with_variants = flow_with_variants.replace(
             "@@prompt_variant_2",
-            post_processing_prompts(json_stringify(top_prompts[2]), _CITATION_TEMPLATE, _USER_INPUT, is_chat))
+            post_processing_prompts(
+                json_stringify(top_prompts[2]), _CITATION_TEMPLATE, _USER_INPUT, is_chat
+            ),
+        )
 
         if USE_CHAT_FLOWS:
-            flow_with_variants = flow_with_variants.replace("@@modify_prompt", _MODIFY_PROMPT)
+            flow_with_variants = flow_with_variants.replace(
+                "@@modify_prompt", _MODIFY_PROMPT
+            )
 
         activity_logger.info(
-            "[Promptflow Creation]: Json payload successfully generated, trying to parse into json dict...")
+            "[Promptflow Creation]: Json payload successfully generated, trying to parse into json dict..."
+        )
         json_payload = json.loads(flow_with_variants)
         activity_logger.info(
-            "[Promptflow Creation]: Json payload successfully parsed, submit to promptflow service now...")
+            "[Promptflow Creation]: Json payload successfully parsed, submit to promptflow service now..."
+        )
 
     ###########################################################################
     # ### construct PF MT service endpoints ### #
-    promptflow_mt_url = SERVICE_ENDPOINT + \
-        "/flow/api" + WORKSPACE_SCOPE + "/flows"
-    headers = get_default_headers(
-        RUN_TOKEN, content_type="application/json")
+    promptflow_mt_url = SERVICE_ENDPOINT + "/flow/api" + WORKSPACE_SCOPE + "/flows"
+    headers = get_default_headers(RUN_TOKEN, content_type="application/json")
 
     response = try_request(promptflow_mt_url, json_payload, headers, activity_logger)
+    if not response.ok:
+        activity_logger.error(
+            f"[Promptflow Creation]: Flow creation failed with Response Code: {response.status_code},"
+            + f"response:{response.text}."
+        )
+        raise Exception(
+            f"Flow creation failed with Response Code: {response.status_code}, response:{response.text}."
+        )
     pf_response_json = json.loads(response.text)
-    flow_id = pf_response_json["flowResourceId"]
-    activity_logger.info("[Promptflow Creation]: Flow creation Succeeded! Id is:" + flow_id, extra={
-        'properties': {'flow_id': flow_id}})
-    run_properties = current_run.get_properties()
-    parent_run_id = run_properties["azureml.pipelinerunid"]
-    parent_run = ws.get_run(parent_run_id)
-    parent_run.add_properties(
-        {"azureml.promptFlowResourceId": flow_id})
-    activity_logger.info("[Promptflow Creation]: Add into run property Succeed! All operation is done!", extra={
-        'properties': {'flow_id': flow_id}})
+    if "flowResourceId" not in pf_response_json:
+        activity_logger.error(
```

Comment:
Should we also throw here? As it stands, this looks like it'll mark the step as being successful; I'm not sure if that's the desired behavior.